app/travel_analytics/package.json

```diff
@@ -12,20 +12,29 @@
     "start": "ui5 serve",
     "test": "npm run test:node && npm run test:java",
     "test:java": "karma start --server=java --single-run",
-    "test:node": "karma start --server=node --single-run"
+    "test:node": "karma start --server=node --single-run",
+    "ts-typecheck": "tsc --noEmit",
+    "prestart": "npm run ts-typecheck",
+    "prebuild": "npm run ts-typecheck"
   },
   "keywords": [
     "ui5",
     "openui5",
     "sapui5"
   ],
   "devDependencies": {
-    "@ui5/cli": "^4.0.0",
     "karma": "^6.4.3",
     "karma-chrome-launcher": "^3.2.0",
     "karma-ui5": "^4.0.1",
     "puppeteer": "^23.0.2",
     "ui5-middleware-simpleproxy": "^3.2.16",
-    "ui5-task-zipper": "^3.1.4"
+    "ui5-task-zipper": "^3.1.4",
+    "@sap/ux-ui5-tooling": "1",
+    "@sapui5/types": "~1.130.0",
+    "ui5-tooling-transpile": "^3.3.7",
+    "typescript": "^5.1.6",
+    "@typescript-eslint/eslint-plugin": "^7.1.1",
+    "@typescript-eslint/parser": "^7.1.1",
+    "@sap-ux/eslint-plugin-fiori-tools": "^0.4.0"
```

Comment:
```suggestion
    "@ui5/cli": "^3.0.0",
    "@sap/ux-ui5-tooling": "1",
    "@sapui5/types": "~1.130.0",
    "ui5-tooling-transpile": "^3.3.7",
    "typescript": "^5.1.6",
    "@typescript-eslint/eslint-plugin": "^7.1.1",
    "@typescript-eslint/parser": "^7.1.1",
    "@sap-ux/eslint-plugin-fiori-tools": "^0.4.0"
```

Reply:
@devinea , is the `ui5-task-zipper` included in the `@ui5/cli`? It's used in the [ui5.yaml](https://github.com/SAP-samples/cap-sflight/blob/4a0f92ba5d719074ac032afab8b33aa2d51a021b/app/travel_analytics/ui5.yaml#L17C13-L17C28)

Reply:
@devinea ui5/cli has been bumped to 4.0 by dependabot 6 months ago:
https://github.com/SAP-samples/cap-sflight/commit/edcc6164698362cf9a49164c4bb0d2cfadd9be73
Are there any issues with version 4, or is version 3 more reliable?

Reply:
> @devinea , is the ui5-task-zipper included in the @ui5/cli? It's used in the [ui5.yaml](https://github.com/SAP-samples/cap-sflight/blob/4a0f92ba5d719074ac032afab8b33aa2d51a021b/app/travel_analytics/ui5.yaml#L17C13-L17C28)

👍 Fine to keep it if already used. npm script `deploy-config` adds it as needed. 

@c-kobo 
Based on https://sap.github.io/ui5-tooling/v4/updates/migrate-v4/ it seems it should be ok for this project. 
There are some potential issues with older ui5 versions such as 1.71 which Fiori tools needs to consider when adding @ui5/cli v4 by default.